packages/ui/src/common/hooks/useSignAndSendTransaction.ts

```diff
@@ -32,32 +33,14 @@ export const useSignAndSendQueryNodeTransaction = ({ transaction, signer, servic
   const sign = useCallback(() => send('SIGN'), [service])
 
   useEffect(() => {
-    if (!isProcessing) {
-      return
-    }
-
-    if (queryNodeStatus === 'confirmed') {
+    if (skipQueryNode && isProcessing) {
       send('SUCCESS')
     }
-  }, [isProcessing, queryNodeStatus])
 
-  return {
-    paymentInfo,
-    sign,
-    isReady,
-  }
-}
-
-export const useSignAndSendTransaction = ({ transaction, signer, service }: Params) => {
-  const { send, paymentInfo, isReady, isProcessing } = useProcessTransaction({ transaction, signer, service })
-
-  const sign = useCallback(() => send('SIGN'), [service])
-
-  useEffect(() => {
-    if (isProcessing) {
+    if (!skipQueryNode && queryNodeStatus === 'confirmed') {
```

Comment:
I think it would be good to move the query refetching:
```ts
apolloClient.refetchQueries({ include: 'active' })
```
in here instead of `@/common/providers/transaction/provider.tsx`, so it only gets called when actually needed

Reply:
Good idea